File: src/3274.check-if-two-chessboard-squares-have-the-same-color.3553/solution.ts
function checkTwoChessboards (coordinate1: string, coordinate2: string): boolean {
    const r1 = coordinate1.charCodeAt(0) - 96;
    const c1 = coordinate1.charCodeAt(1) - 48;
    const r2 = coordinate2.charCodeAt(0) - 96;
    const c2 = coordinate2.charCodeAt(1) - 48;

    const isBlack1 = r1 % 2 === c1 % 2;
    const isBlack2 = r2 % 2 === c2 % 2;

    return isBlack1 === isBlack2;
}

File: src/3274.check-if-two-chessboard-squares-have-the-same-color.3553/solution.java
class Solution {
    public boolean checkTwoChessboards(String coordinate1, String coordinate2) {
        int r1 = coordinate1.charAt(0)- 'a';
        int c1 = coordinate1.charAt(1)- '0';
        boolean isWhite1 = r1%2 == c1%2;

        int r2 = coordinate2.charAt(0)- 'a';
        int c2 = coordinate2.charAt(1)- '0';
        boolean isWhite2 = r2%2 == c2%2;  
        
        return isWhite1 == isWhite2;
    }
}